### PR TITLE
JIRA: VZ-7361 enable MySQL backup job cleanup in VPO

### PR DESCRIPTION
**Pull request by jmaron99**
VPO will watch for new job creation and cleanup any backup jobs created in keycloak namespace once the job completes.

### CODE REVIEW

File: platform-operator/controllers/verrazzano/controller.go [hunk 1]
```diff
@@ -518,6 +529,63 @@ func (r *Reconciler) cleanupUninstallJob(jobName string, namespace string, log v
...
 
+// cleanupMysqlBackupJob checks for the existence of a stale MySQL restore job and deletes the job if one is found
+func (r *Reconciler) cleanupMysqlBackupJob(log vzlog.VerrazzanoLogger) error {
+	// Check if jobs for running the restore jobs exist
+	jobsFound := &batchv1.JobList{}
+	log.Debug("Checking if stale restore jobs exist")
+	createByReq, _ := kblabels.NewRequirement("app.kubernetes.io/created-by", selection.Equals, []string{constants.MySQLOperator})
+	labelSelector := kblabels.NewSelector()
+	labelSelector = labelSelector.Add(*createByReq)
+	err := r.List(context.TODO(), jobsFound, &client.ListOptions{LabelSelector: labelSelector})
```
Comment (theabmitra) [inline platform-operator/controllers/verrazzano/controller.go:540]: I think Jobs spun off from Cron-Jobs (schedule) will not have these labels and will be not caught i think

File: platform-operator/apis/verrazzano/webhooks/mysql_backup_webhook.go [hunk 2]
```diff
@@ -211,7 +211,7 @@ func (m *MySQLBackupJobWebhook) isCronJobCreatedByMysqlOperator(req admission.Re
 		}
 	}
 
-	if mysqlOperatorOwnerReferencePresent && mysqlOperatorLabelPresent {
+	if mysqlOperatorOwnerReferencePresent || mysqlOperatorLabelPresent {
```
Comment (theabmitra) [inline platform-operator/apis/verrazzano/webhooks/mysql_backup_webhook.go:214]: This should be AND i feel ?

Comment (jmaron99) [reply platform-operator/apis/verrazzano/webhooks/mysql_backup_webhook.go:214]: I found in my testing that the cron job, for example, did not have an ownership reference to the operator so this check was failing.  I feel this is safer given that finding.

Comment (jmaron99) [reply platform-operator/apis/verrazzano/webhooks/mysql_backup_webhook.go:214]: `kubectl get cronjob -n keycloak -o yaml
apiVersion: v1
items:
- apiVersion: batch/v1
  kind: CronJob
  metadata:
    creationTimestamp: "2022-10-24T14:55:01Z"
    labels:
      app.kubernetes.io/created-by: mysql-operator
      app.kubernetes.io/instance: idc-mysql
      app.kubernetes.io/managed-by: mysql-operator
      app.kubernetes.io/name: mysql-innodbcluster
      mysql.oracle.com/cluster: mysql
      tier: mysql
    name: mysql-test-schedule-cb
    namespace: keycloak
    resourceVersion: "200515"
    uid: c8612326-c2fa-449b-a65c-cb94e9b7245b
  spec:
    concurrencyPolicy: Forbid
    failedJobsHistoryLimit: 1
    jobTemplate:
      metadata:
        creationTimestamp: null
      spec:
        backoffLimit: 0
        template:
          metadata:
            creationTimestamp: null
          spec:
            containers:
            - command:
              - mysqlsh
              - --pym
              - mysqloperator
              - backup
              - --command
              - create-backup-object
              - --namespace
              - keycloak
              - --cluster-name
              - mysql
              - --schedule-name
              - test-schedule
              env:
              - name: MYSQLSH_USER_CONFIG_HOME
                value: /mysqlsh
              image: ghcr.io/verrazzano/mysql-operator:8.0.31-2.0.7
              imagePullPolicy: IfNotPresent
              name: operator-backup-job-cron
              resources: {}
              securityContext:
                runAsUser: 27
              terminationMessagePath: /dev/termination-log
              terminationMessagePolicy: File
              volumeMounts:
              - mountPath: /mysqlsh
                name: shellhome
            dnsPolicy: ClusterFirst
            restartPolicy: Never
            schedulerName: default-scheduler
            securityContext: {}
            serviceAccount: mysql-sa
            serviceAccountName: mysql-sa
            terminationGracePeriodSeconds: 60
            volumes:
            - emptyDir: {}
              name: shellhome
    schedule: 0,23,36,56 * * * *
    successfulJobsHistoryLimit: 3
    suspend: false
  status:
    lastScheduleTime: "2022-10-24T21:36:00Z"
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""`

Comment (jmaron99) [reply platform-operator/apis/verrazzano/webhooks/mysql_backup_webhook.go:214]: `
kubectl get cronjob -n keycloak -o yaml
apiVersion: v1
items:
- apiVersion: batch/v1
  kind: CronJob
  metadata:
    creationTimestamp: "2022-10-24T14:55:01Z"
    labels:
      app.kubernetes.io/created-by: mysql-operator
      app.kubernetes.io/instance: idc-mysql
      app.kubernetes.io/managed-by: mysql-operator
      app.kubernetes.io/name: mysql-innodbcluster
      mysql.oracle.com/cluster: mysql
      tier: mysql
    name: mysql-test-schedule-cb
    namespace: keycloak
    resourceVersion: "200515"
    uid: c8612326-c2fa-449b-a65c-cb94e9b7245b
  spec:
    concurrencyPolicy: Forbid
    failedJobsHistoryLimit: 1
    jobTemplate:
      metadata:
        creationTimestamp: null
      spec:
        backoffLimit: 0
        template:
          metadata:
            creationTimestamp: null
          spec:
            containers:
            - command:
              - mysqlsh
              - --pym
              - mysqloperator
              - backup
              - --command
              - create-backup-object
              - --namespace
              - keycloak
              - --cluster-name
              - mysql
              - --schedule-name
              - test-schedule
              env:
              - name: MYSQLSH_USER_CONFIG_HOME
                value: /mysqlsh
              image: ghcr.io/verrazzano/mysql-operator:8.0.31-2.0.7
              imagePullPolicy: IfNotPresent
              name: operator-backup-job-cron
              resources: {}
              securityContext:
                runAsUser: 27
              terminationMessagePath: /dev/termination-log
              terminationMessagePolicy: File
              volumeMounts:
              - mountPath: /mysqlsh
                name: shellhome
            dnsPolicy: ClusterFirst
            restartPolicy: Never
            schedulerName: default-scheduler
            securityContext: {}
            serviceAccount: mysql-sa
            serviceAccountName: mysql-sa
            terminationGracePeriodSeconds: 60
            volumes:
            - emptyDir: {}
              name: shellhome
    schedule: 0,23,36,56 * * * *
    successfulJobsHistoryLimit: 3
    suspend: false
  status:
    lastScheduleTime: "2022-10-24T21:36:00Z"
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""

`

**Review by theabmitra — APPROVED** [review]
Approved

**Pull request merged**